src/migration/cassandra/CassandraMigrationBackend.hpp

```diff
@@ -0,0 +1,127 @@
+
+namespace migration::cassandra {
+
+/**
+ * @brief The backend for the migration. It is a subclass of the CassandraBackend and provides the migration specific
+ * functionalities.
+ */
+class CassandraMigrationBackend : public data::cassandra::CassandraBackend {
+    util::Logger log_{"Migration"};
+    data::cassandra::SettingsProvider settingsProvider_;
+    CassandraMigrationSchema migrationSchema_;
+
+public:
+    /**
+     * @brief Construct a new Cassandra Migration Backend object. The backend is not readonly.
+     *
+     * @param settingsProvider The settings provider
+     */
+    explicit CassandraMigrationBackend(data::cassandra::SettingsProvider settingsProvider)
+        : data::cassandra::CassandraBackend{auto{settingsProvider}, false /* not readonly */}
+        , settingsProvider_(std::move(settingsProvider))
+        , migrationSchema_{settingsProvider_}
+    {
+    }
+
+    /**
+     *@brief Scan a table in a token range and call the callback for each row
+     *
+     *@tparam TableDesc The table description of the table to scan
+     *@param start The start token
+     *@param end The end token
+     *@param callback The callback to call for each row
+     *@param yield The boost asio yield context
+     */
+    template <TableSpec TableDesc>
+    void
+    migrateInTokenRange(
+        std::int64_t const& start,
+        std::int64_t const& end,
+        auto const& callback,
+        boost::asio::yield_context yield
+    )
+    {
+        LOG(log_.debug()) << "Travsering token range: " << start << " - " << end
+                          << "- table: " << TableDesc::TABLE_NAME;
+        // for each table we only have one prepared statement
+        static auto statementPrepared =
+            migrationSchema_.getPreparedFullScanStatement(handle_, TableDesc::TABLE_NAME, TableDesc::PARTITION_KEY);
+
+        auto const statement = statementPrepared.bind(start, end);
+
+        auto const res = this->executor_.read(yield, statement);
+        auto const& results = res.value();
+        if (not results.hasRows()) {
+            LOG(log_.debug()) << "No rows returned  - table: " << TableDesc::TABLE_NAME << " - range: " << start
+                              << " - " << end;
+            return;
+        }
+
+        for (auto const& row : std::apply(
+                 [&](auto... args) { return data::cassandra::extract<decltype(args)...>(results); },
+                 typename TableDesc::Row{}
+             )) {
+            callback(row);
+        }
+    }
+
+    /**
+     *@brief Record the migration status of a migrator in the database
+     *
+     *@param migratorName The name of the migrator
+     */
+    void
+    writeMigratedMigrator(std::string const& migratorName)
+    {
+        auto const& preparedStatement = migrationSchema_.getPreparedInsertMigratedMigrator(handle_);
+        this->executor_.writeSync(
+            preparedStatement, data::cassandra::Text("migrated"), data::cassandra::Text{migratorName}
```

Comment:
The second arg is the name of the migration. The first arg is the status. To avoid the full table scan, I need a partition key I know to do the query.